Quadriga/src/main/java/edu/asu/spring/quadriga/web/workbench/PublicPageController.java

```diff
@@ -0,0 +1,70 @@
+package edu.asu.spring.quadriga.web.workbench;
+
+import java.util.Properties;
+import javax.annotation.Resource;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.servlet.ModelAndView;
+import edu.asu.spring.quadriga.aspects.annotations.AccessPolicies;
+import edu.asu.spring.quadriga.aspects.annotations.CheckedElementType;
+import edu.asu.spring.quadriga.aspects.annotations.ElementAccessPolicy;
+import edu.asu.spring.quadriga.domain.factory.workbench.IPublicPageFactory;
+import edu.asu.spring.quadriga.exceptions.QuadrigaAccessException;
+import edu.asu.spring.quadriga.exceptions.QuadrigaStorageException;
+import edu.asu.spring.quadriga.web.login.RoleNames;
+
+@Controller
+public class PublicPageController {
+
+    @Autowired
+    private IPublicPageFactory publicPageFactory;
+
+    @Resource(name = "projectconstants")
+    private Properties messages;
+
+    /**
+     * This method is called during the load of Public page settings form
+     *
+     * @return model - model object
+     */
+    @AccessPolicies({ @ElementAccessPolicy(type = CheckedElementType.PROJECT, paramIndex = 1, userRole = {
+            RoleNames.ROLE_COLLABORATOR_ADMIN,
+            RoleNames.ROLE_PROJ_COLLABORATOR_ADMIN }) })
+    @RequestMapping(value = "auth/workbench/{projectid}/addpublicpage", method = RequestMethod.GET)
+    public ModelAndView publicPageSettingsForm(
+            @PathVariable("projectid") String projectid)
+            throws QuadrigaStorageException, QuadrigaAccessException {
+        ModelAndView model = new ModelAndView("auth/workbench/addpublicpage");
+        model.getModelMap().put("publicpage",
+                publicPageFactory.createPublicPageObject());
+        model.getModelMap().put("ppprojectid", projectid);
+        return model;
+    }
+
+    /**
+     * This method is used update the Text blocks in the Public settings page
+     *
+     * @return json
+     */
+
+    @RequestMapping(method = RequestMethod.GET, value = "auth/workbench/{projectid}/addpublicpage1")
+    public @ResponseBody ResponseEntity<String> getExistingConcepts(
+            @RequestParam("data1") String data1)

```

Comment:
this data will contain the the updated information about the text blocks which will be stored in database(QUAD-111)